models/error.go

```diff
@@ -1371,6 +1372,48 @@ func (err ErrMergePushOutOfDate) Error() string {
 	return fmt.Sprintf("Merge PushOutOfDate Error: %v: %s\n%s", err.Err, err.StdErr, err.StdOut)
 }
 
+// ErrPushRejected represents an error if merging fails due to unrelated histories
+type ErrPushRejected struct {
+	Style  MergeStyle
+	StdOut string
+	StdErr string
+	Err    error
+}
+
+// IsErrPushRejected checks if an error is a ErrPushRejected.
+func IsErrPushRejected(err error) bool {
+	_, ok := err.(ErrPushRejected)
+	return ok
+}
+
+func (err ErrPushRejected) Error() string {
+	return fmt.Sprintf("Merge PushRejected Error: %v: %s\n%s", err.Err, err.StdErr, err.StdOut)
+}
+
+// Message returns the cleaned message from Error
+func (err ErrPushRejected) Message() string {
+	messageBuilder := &strings.Builder{}
+	i := 0
+	for {
+		if len(err.StdErr) <= i+8 {
+			break
+		}
+		if err.StdErr[i:i+8] != "remote: " {
+			break
```

Comment:
I don't understand this. If the message doesn't start with "remote: ", the rest of the loop will never execute. Is that intended? What's a typical message here?

Reply:
We are deliberately only extracting the remote message here. The rest of the git bumpf is irrelevant.

Reply:
I mean, shoudln't that `break` be `i++` and `continue` instead? The message I get doesn't even contain `"remote: "`:

```
Failed to update/create file 'vaafallar' with error: Unable to push back to repo from temporary repo: user1/publicrepo (/home/gittest/data/tmp/local-repo/upload.git026502734) Error: exit status 1 - To /home/gittest/gitea-repositories/user1/publicrepo.git
 ! [remote rejected] 3c35192d4bf4c5fb936309f4d56cd5c203d37297 -> master (pre-receive hook declined)
error: failed to push some refs to '/home/gittest/gitea-repositories/user1/publicrepo.git'
```

And from command line:
```
$ git push
Enumerating objects: 3, done.
Counting objects: 100% (3/3), done.
Delta compression using up to 2 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (2/2), 222 bytes | 222.00 KiB/s, done.
Total 2 (delta 1), reused 0 (delta 0)
To git.mydomain.com:user1/publicrepo.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'gittest@git.mydomain.com:user1/publicrepo.git'
```

Reply:
Then your git hook is just rejecting without returning a message contrary to good practice.

Reply:
I've adjusted the return.

Reply:
> 
> 
> Then your git hook is just rejecting without returning a message contrary to good practice.

Right. My bad. 😄